authentikos/README.md

```diff
@@ -0,0 +1,30 @@
+# Authentikos ([αὐθεντικός](https://en.wikipedia.org/wiki/Authentication))
+
+## Description
+
+`authentikos` is a service used to create/refresh a Google oauth token and store the value in a Kubernetes secret.
```

Comment:
a) Why not use https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity?
b) Why not put it here rather than in k/test-infra?

Reply:
> Why not use https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity?

We actually need to physically pass the oauth token down to the container code for a simple call. There is a make recipe that [performs a `curl` to `https://storage.googleapis.com/`](https://github.com/istio/istio/blob/53d1eefbdd927e723bc4ec4f17065d15c599af80/bin/init.sh#L95) and an access token needs to be made available via the environment as an `Authorization` header. I don't think [workload-identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) can help in this case?

Reply:
Please use workload identity to get the access token, rather than relying on a json file that can be stolen.

https://cloud.google.com/kubernetes-engine/docs/how-to/protecting-cluster-metadata#verify-concealed

https://cloud.google.com/compute/docs/storing-retrieving-metadata
https://cloud.google.com/compute/docs/instances/verifying-instance-identity#request_signature

Reply:
It's the first one: `All the clients in sub-packages support authentication via Google Application Default Credentials`

https://godoc.org/google.golang.org/api#hdr-Authentication_and_Authorization

Reply:
If you enable workload identity in your pod the client will automatically detect and use those credentials.

Reply:
Currently, the way this works it that if a `--creds` file is **not** supplied then it will use *Google application default credentials*. That is how I have it [setup](https://github.com/istio/test-infra/pull/2106/files#diff-b7414ef890d889f22c3913f1c18fee2dR166-R186). It sounds like I just need to enable this in the nodes ... but I do not think it is a code change to allow this in the service.

Reply:
Sounds great!

Reply:
I was able to verify that in a cluster with **workload-identity** enabled, the Google creds client will pick up the credentials from the metadata server. 🍾